lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapComponent.kt

```diff
@@ -93,10 +98,10 @@ fun W3WMapComponent(
     layoutConfig: W3WMapDefaults.LayoutConfig = W3WMapDefaults.defaultLayoutConfig(),
     mapConfig: W3WMapDefaults.MapConfig = W3WMapDefaults.defaultMapConfig(),
     mapState: W3WMapState,
-    locationSource: W3WLocationSource? = null,
```

Comment:
Since we are exposing the mapState, the client already has control over the composable. Therefore, the locationSource parameter is unnecessary. Instead, we provide a callback to notify when the MyLocation button is clicked.